### PR TITLE
Add support for KBDPAD MKII

### PR DESCRIPTION
**Pull request by ai03-2725**
## Description

Adds support for the KBDPAD MKII numpad.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Adds the KBDPAD MKII

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kbdfans/kbdpad_mk2/kbdpad_mk2.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+/* This a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT( \
+    K00, K01, K02, K03, \
```
Comment (mechmerlin) [inline keyboards/kbdfans/kbdpad_mk2/kbdpad_mk2.h]: curious why you decided to go Kxy here and for the kbd8x mkii you went with Kxyz.

File: keyboards/kbdfans/kbdpad_mk2/config.h [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+// /*== customize breathing effect ==*/
+//   /*==== (DEFAULT) use fixed table instead of exp() and sin() ====*/
+//   #define RGBLIGHT_BREATHE_TABLE_SIZE 256      // 256(default) or 128 or 64
+//   /*==== use exp() and sin() ====*/
+//   #define RGBLIGHT_EFFECT_BREATHE_CENTER 1.85  // 1 to 2.7
+//   #define RGBLIGHT_EFFECT_BREATHE_MAX    255   // 0 to 255
+#endif
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCING_DELAY 5
```
Comment (e11i0t23) [inline keyboards/kbdfans/kbdpad_mk2/config.h]:
```suggestion
#define DEBOUNCE 5
```
`DEBOUNCING_DELAY` has been depricated in favour of `DEBOUNCE`

File: keyboards/kbdfans/kbdpad_mk2/rules.mk [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# MCU name
+#MCU = at90usb1286
```
Comment (e11i0t23) [inline keyboards/kbdfans/kbdpad_mk2/rules.mk]:
```suggestion
```

File: keyboards/kbdfans/kbdpad_mk2/rules.mk [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+#   Atmel DFU loader    4096
+#   LUFA bootloader     4096
+#   USBaspLoader        2048
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no      # Virtual DIP switch configuration(+1000)
```
Comment (e11i0t23) [inline keyboards/kbdfans/kbdpad_mk2/rules.mk]:
```suggestion
BOOTMAGIC_ENABLE = lite    # Virtual DIP switch configuration(+1000)
```

Adds an easier way for people to enter reset as i don't see easy access to reset switch

File: keyboards/kbdfans/kbdpad_mk2/kbdpad_mk2.h [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+/* This a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT( \
```
Comment (drashna) [inline keyboards/kbdfans/kbdpad_mk2/kbdpad_mk2.h]:
```suggestion
#define LAYOUT_ortho_6x4( \
```

File: keyboards/kbdfans/kbdpad_mk2/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT( /* Base */
```
Comment (drashna) [inline keyboards/kbdfans/kbdpad_mk2/keymaps/default/keymap.c]:
```suggestion
  [0] = LAYOUT_ortho_6x4( /* Base */
```

File: keyboards/kbdfans/kbdpad_mk2/keymaps/tester/keymap.c [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT( /* Base */
```
Comment (drashna) [inline keyboards/kbdfans/kbdpad_mk2/keymaps/tester/keymap.c]:
```suggestion
  [0] = LAYOUT_ortho_6x4( /* Base */
```

File: keyboards/kbdfans/kbdpad_mk2/info.json [hunk 8]
```diff
@@ -0,0 +1,37 @@
+{
+    "keyboard_name": "kbdpad_mk2", 
+    "url": "https://kb.ai03.me/projects/kbdpad-mkii.html", 
+    "maintainer": "ai03", 
+    "width": 4, 
+    "height": 6.25, 
+    "layouts": {
+        "LAYOUT": {
```
Comment (drashna) [inline keyboards/kbdfans/kbdpad_mk2/info.json]:
```suggestion
        "LAYOUT_ortho_6x4": {
```

File: keyboards/kbdfans/kbdpad_mk2/kbdpad_mk2.h (hunk 1 above)
Comment (ai03-2725) [reply keyboards/kbdfans/kbdpad_mk2/kbdpad_mk2.h]: Looking back at it, I have no idea. Probably either extreme fatigue, or too long of a gap between the pad and 8x's revised PCB that I forgot how I labeled it on the 8x.

File: keyboards/kbdfans/kbdpad_mk2/rules.mk [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
```
Comment (drashna) [inline keyboards/kbdfans/kbdpad_mk2/rules.mk]: I missed one thing, sorry! 

```suggestion
HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)

LAYOUTS = ortho_6x4
```

**Pull request merged**